parser_library/src/workspaces/workspace.cpp

```diff
@@ -572,53 +618,69 @@ std::optional<std::string> substitute_home_directory(std::string p)
+    {
+        opts.extensions = macro_extensions_compatibility_list;
+        opts.extensions_from_deprecated_source = true;
+    }
+
+    return opts;
+}
+
+// Constructs resource location of a library and analyzes if it contains wildcards
+std::pair<utils::resource::resource_location, bool> construct_and_analyze_lib_resource_location(
+    const std::string& lib_path, const utils::resource::resource_location& base)
+{
+    auto asterisk = lib_path.find('*');
+    auto last_valid_slash = lib_path.find_last_of("/\\", asterisk);
+    utils::resource::resource_location rl;
+
+    if (asterisk != std::string::npos && last_valid_slash != std::string::npos)
+    {
+        // This is a path with an asterisk
+        // Split the path at the last slash before the asterisk, transform the first part into resource_location, then
+        // join the second part
+        rl = transform_to_resource_location(lib_path.substr(0, last_valid_slash), base);
+        rl.join(lib_path.substr(last_valid_slash + 1));
+    }
+    else
+        rl = transform_to_resource_location(lib_path, base);
+
+    rl.join(""); // Ensure that this is a directory
+    return std::make_pair(utils::resource::resource_location(rl.lexically_normal()), asterisk != std::string::npos);
+}
 } // namespace
 
 void workspace::process_processor_group(
     const config::processor_group& pg, const config::proc_grps& proc_groups, const config::pgm_conf& pgm_config)
 {
-    std::filesystem::path ws_path(location_.get_path());
-
     processor_group prc_grp(pg.name, pg.asm_options, pg.preprocessor);
 
     for (const auto& lib : pg.libs)
     {
-        std::filesystem::path lib_path = [&path = lib.path]() {
-            if (!path.empty())
-                return utils::path::join(path, "");
-            return std::filesystem::path {};
-        }();
-        if (auto sp = substitute_home_directory(lib_path.string()); sp.has_value())
-            lib_path = std::move(sp.value());
-        else
-        {
-            config_diags_.push_back(diagnostic_s::warning_L0006(lib_path.string()));
-            continue;
-        }
-        if (!utils::path::is_absolute(lib_path))
-            lib_path = utils::path::join(ws_path, lib_path);
-        lib_path = utils::path::lexically_normal(lib_path);
-
-        library_local_options opts;
-        opts.optional_library = lib.optional;
-        if (!lib.macro_extensions.empty())
-            opts.extensions = lib.macro_extensions;
-        else if (!proc_groups.macro_extensions.empty())
-            opts.extensions = proc_groups.macro_extensions;
-        else if (auto macro_extensions_compatibility_list = get_macro_extensions_compatibility_list(pgm_config);
-                 !macro_extensions_compatibility_list.empty())
-        {
-            opts.extensions = macro_extensions_compatibility_list;
-            opts.extensions_from_deprecated_source = true;
-        }
+        auto lib_local_opts = get_library_local_options(lib, proc_groups, pgm_config);
```

Comment:
[missing call to substitute_home_directory](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/blob/ebc1ae8b1a9462e8f9856b99f3b1966a256a39b4/parser_library/src/workspaces/workspace.cpp#L591)